doc/src/manual/multi-threading.md

```diff
@@ -239,6 +239,75 @@ julia> a
 
 Note that [`Threads.@threads`](@ref) does not have an optional reduction parameter like [`@distributed`](@ref).
 
+### Using `@threads` without data races
+
+Taking the example of a naive sum
+
+```julia-repl
+julia> function sum_single(a)
+           s = 0
+           for i in a
+               s += i
+           end
+           s
+       end
+sum_single (generic function with 1 method)
+
+julia> sum_single(1:1_000_000)
+500000500000
+```
+
+Simply adding `@threads` exposes a data race with multiple threads reading and writing `s` at the same time.
+```julia-repl
+julia> function sum_multi_bad(a)
+           s = 0
+           Threads.@threads for i in a
+               s += i
+           end
+           s
+       end
+sum_multi_bad (generic function with 1 method)
+
+julia> sum_multi_bad(1:1_000_000)
+70140554652
+```
+
+Note that the result is not `500000500000` as it should be, and will most likely change each evaluation.
+
+To fix this, buffers that are specific to the task may be used to segment the sum into chunks that are race-free.
+
+```julia-repl
+julia> function sum_multi_good(a)
+           chunks = Iterators.partition(a, length(a) ÷ Threads.nthreads())
+           tasks = map(chunks) do chunk
+               Threads.@spawn begin
+                   b = zero(eltype(chunk))
+                   for i ∈ chunk
+                       b += i
+                   end
+                   return b
+               end
+           end
+           s = 0
+           buffers = fetch.(tasks)
+           for b in buffers
+               s += b
+           end
+           s
+       end
+sum_multi_good (generic function with 1 method)
+
+julia> sum_multi_good(1:1_000_000)
+500000500000
+```
+
+Note buffers should not be managed based on the `threadid()` i.e. `buffers = zeros(Threads.nthreads())` because tasks can
+actually change thread at yield points, known as [task migration](@ref man-task-migration), which introduces potential for
+data races for such a buffering approach.
```

Comment:
I'm not very clear on the lingo here, but I think the main concern is actually `yield`ing, not migration -- or do we consider `yield` a form of migration even if the `threadid` doesn't change?

Reply:
Yeah. Should be fixed now